src/domain/opendatafabric/src/identity/dataset_refs.rs

```diff
@@ -740,3 +741,126 @@ impl std::cmp::PartialOrd for DatasetRefAny {
         Some(self.cmp(other))
     }
 }
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum DatasetRefPattern {
+    Ref(DatasetRef),
+    Pattern(Option<AccountName>, DatasetNamePattern),
+}
+
+impl DatasetRefPattern {
+    pub fn match_pattern(dataset_ref: &str, pattern: &str) -> Result<bool, InvalidPatternError> {
```

Comment:
The type is already called `DatasetRefPattern`, so calling the method `match_pattern` is redundant. I propose just `matches()` or `is_match()` for similarity with [regex crate](https://docs.rs/regex/latest/regex/struct.Regex.html#method.is_match).

Reply:
Also why are we accepting `dataset_ref` as `str` and not `DatasetRef`? I think this can lead to multiple problems.